src/orange/univ/detail/LogicEvaluator.hh

```diff
@@ -59,6 +65,17 @@ CELER_FUNCTION LogicEvaluator::LogicEvaluator(SpanConstLogic logic)
  * Evaluate a logical expression, substituting bools from the sense view.
  */
 CELER_FUNCTION bool LogicEvaluator::operator()(SpanConstSense values) const
+{
+    auto calc_sense = [&](FaceId face_id) { return values[face_id.get()]; };
+    return (*this)(calc_sense);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Evaluate a logical expression, with on-the-fly sense evaluation.
+ */
+template<class F, std::enable_if_t<std::is_invocable_v<F, FaceId>, bool>>
```

Comment:
Neat!

Reply:
this is actually necessary otherwise this overload is a better match than `SpanConstSense` in some situations.